src/main/java/com/google/devtools/build/lib/bazel/bzlmod/Discovery.java

```diff
@@ -60,7 +60,7 @@ public static Result run(Environment env, RootModuleFileValue root)
       throws InterruptedException, ExternalDepsException {
     String rootModuleName = root.getModule().getName();
     ImmutableMap<String, ModuleOverride> overrides = root.getOverrides();
-    Map<ModuleKey, InterimModule> depGraph = new HashMap<>();
+    Map<ModuleKey, InterimModule> depGraph = new LinkedHashMap<>();
```

Comment:
just a comment, no action to take: I don't think this one really matters (by my read, at most an error message could be different).

Reply:
Most (or even all) of the other findings also don't affect build output or correctness, but I am also aiming for deterministic error messages (assuming `--keep_going`, as which error is hit first is inherently non-deterministic).